metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -1318,7 +1333,22 @@ class Compilers(
       targetId: BuildTargetIdentifier
   ): Option[PresentationCompiler] =
     withKeyAndDefault(targetId) { case (key, getCompiler) =>
-      Option(jcache.computeIfAbsent(key, { _ => getCompiler() }).await)
+      val sources = buildTargets.buildTargetSources(targetId)
+      val modifiedFiles =
```

Comment:
Is it modified files or any open files within a target?  In case of the first option OutlineFilesProvider already has that data.